packages/pg-gateway/src/connection.ts

```diff
@@ -137,11 +137,13 @@ export default class PostgresConnection {
   hasStarted = false;
   isAuthenticated = false;
   detached = false;
-  writer = new BufferWriter();
-  reader = new BufferReader();
+  bufferWriter = new BufferWriter();
+  bufferReader = new BufferReader();
   clientParams?: ClientParameters;
   tlsInfo?: TlsInfo;
   messageBuffer = new MessageBuffer();
+  // reference to the stream writer when processing data
+  streamWriter?: WritableStreamDefaultWriter<Uint8Array>;
```

Comment:
we are using a reference to the writer in `processData` so external processes can also write into the stream (like my websocket server in this case).